flink-python/pyflink/fn_execution/coder_impl.py

```diff
@@ -373,3 +376,45 @@ def internal_to_timestamp(self, milliseconds, nanoseconds):
         second, microsecond = (milliseconds // 1000,
                                milliseconds % 1000 * 1000 + nanoseconds // 1000)
         return datetime.datetime.utcfromtimestamp(second).replace(microsecond=microsecond)
+
+
+class ArrowCoderImpl(StreamCoderImpl):
+
+    def __init__(self, schema):
+        self._schema = schema
+        self._resettable_io = ResettableIO()
+
+    def encode_to_stream(self, cols, out_stream, nested):
+        if not hasattr(self, "_batch_writer"):
+            self._batch_writer = pa.RecordBatchStreamWriter(self._resettable_io, self._schema)
+
+        self._resettable_io.set_output_stream(out_stream)
+        self._batch_writer.write_batch(self._create_batch(cols))
+
+    def decode_from_stream(self, in_stream, nested):
+        if not hasattr(self, "_batch_reader"):
```

Comment:
I'm wondering if we can avoid these `hasattr`. Can we use `pa.ipc.open_stream` read the stream directly and return the batch?